wolfcrypt/src/integer.c

```diff
@@ -868,16 +868,13 @@ int mp_exptmod (mp_int * G, mp_int * X, mp_int * P, mp_int * Y)
      return MP_VAL;
   }
   if (mp_isone(P)) {
-     mp_set(Y, 0);
-     return MP_OKAY;
```

Comment:
With the code change added it will still return early, but instead of always returning MP_OKAY it returns the result of mp_set. Which can be MP_MEM (if mp_grow fails) or MP_OKAY if mp_set is successful. This fixes the case where mp_set could fail with MP_MEM and the resulting failure would not be caught.